browser/brave_wallet/keyring_service_unittest.cc

```diff
@@ -2556,7 +2560,185 @@ TEST_F(KeyringServiceUnitTest, ImportFilecoinAccounts) {
+  base::RunLoop().RunUntilIdle();
+  TestKeyringServiceObserver observer;
+  service.AddObserver(observer.GetReceiver());
+  EXPECT_TRUE(
+      HasPrefForKeyring(kPasswordEncryptorSalt, mojom::kDefaultKeyringId));
+  EXPECT_TRUE(
+      HasPrefForKeyring(kPasswordEncryptorNonce, mojom::kDefaultKeyringId));
+  EXPECT_TRUE(HasPrefForKeyring(kEncryptedMnemonic, mojom::kDefaultKeyringId));
+
+  EXPECT_TRUE(
+      HasPrefForKeyring(kPasswordEncryptorSalt, mojom::kFilecoinKeyringId));
+  EXPECT_FALSE(
+      HasPrefForKeyring(kPasswordEncryptorNonce, mojom::kFilecoinKeyringId));
+  EXPECT_FALSE(
+      HasPrefForKeyring(kEncryptedMnemonic, mojom::kFilecoinKeyringId));
+  EXPECT_FALSE(service.IsKeyringExist(mojom::kFilecoinKeyringId));
+  EXPECT_TRUE(service.IsKeyringExist(mojom::kDefaultKeyringId));
+  const struct {
+    const char* name;
+    const char* private_key;
+    const char* address;
+    const char* type;
+    const char* public_key;
+  } imported_accounts[] = {
+      {"Imported Filecoin account 1",
+       /*"rQG5jnbc+y64fckG+T0EHVwpLBmW9IgAT7U990HXcGk=",*/
+       "ad01b98e76dcfb2eb87dc906f93d041d5c292c1996f488004fb53df741d77069",
+       "t1lqarsh4nkg545ilaoqdsbtj4uofplt6sto26ziy", "secp256k1", ""}};
+
+  bool callback_called = false;
+  service.ImportFilecoinSECP256K1Account(
+      imported_accounts[0].name, imported_accounts[0].private_key,
+      mojom::kFilecoinTestnet,
+      base::BindLambdaForTesting([&](bool success, const std::string& address) {
+        EXPECT_TRUE(success);
+        EXPECT_EQ(imported_accounts[0].address, address);
+        callback_called = true;
+      }));
+  base::RunLoop().RunUntilIdle();
+  ASSERT_TRUE(callback_called);
+  EXPECT_TRUE(observer.IsKeyringCreated(mojom::kFilecoinKeyringId));
+
+  callback_called = false;
+  // Imported accounts should be restored
+  service.GetKeyringInfo(
+      brave_wallet::mojom::kFilecoinKeyringId,
+      base::BindLambdaForTesting([&](mojom::KeyringInfoPtr keyring_info) {
+        EXPECT_EQ(keyring_info->account_infos.size(), 1u);
+        EXPECT_EQ(keyring_info->account_infos[0]->address,
+                  imported_accounts[0].address);
+        EXPECT_EQ(keyring_info->account_infos[0]->name,
+                  imported_accounts[0].name);
+        EXPECT_TRUE(keyring_info->account_infos[0]->is_imported);
+        callback_called = true;
+      }));
+  base::RunLoop().RunUntilIdle();
+  EXPECT_TRUE(callback_called);
+  EXPECT_NE(
+      GetStringPrefForKeyring(kEncryptedMnemonic, mojom::kDefaultKeyringId),
+      GetStringPrefForKeyring(kEncryptedMnemonic, mojom::kFilecoinKeyringId));
+  EXPECT_TRUE(service.IsKeyringExist(mojom::kFilecoinKeyringId));
+  EXPECT_TRUE(service.IsKeyringExist(mojom::kDefaultKeyringId));
+  auto default_mnemonic =
+      service.GetMnemonicForKeyringImpl(mojom::kDefaultKeyringId);
+  ASSERT_FALSE(default_mnemonic.empty());
+  EXPECT_EQ(default_mnemonic,
+            service.GetMnemonicForKeyringImpl(mojom::kFilecoinKeyringId));
```

Comment:
Don't add back the test cases that I removed in https://github.com/brave/brave-core/commit/7db10e913cce47efa1140a75b6d4c2b8cf552ff4

Reply:
I see, done, thanks!